cpp/src/arrow/acero/hash_join_node.cc

```diff
@@ -53,6 +53,9 @@ bool HashJoinSchema::IsTypeSupported(const DataType& type) {
   if (id == Type::EXTENSION) {
     return IsTypeSupported(*checked_cast<const ExtensionType&>(type).storage_type());
   }
+  if (id == Type::LIST) {
```

Comment:
should we also support the other list types under `is_list_like`? probably not map, but it seems like large list and fixed size list should just work?